### PR TITLE
Feature: Added filters to integrate related posts

### PR DESCRIPTION
**Pull request by JeanPaiva**
Adding some filters to facilitate integration with GBP

### CODE REVIEW

File: src/blocks/query-loop/components/InspectorControls.js [hunk 1]
```diff
@@ -28,10 +27,8 @@ export default ( { attributes, setAttributes } ) => {
 	}, [] );
 
 	useEffect( () => {
-		if ( ! isEmpty( queryState ) && ! isEqual( attributes.query, queryState ) ) {
-			setAttributes( { query: queryState } );
-		}
-	}, [ queryState ] );
+		setAttributes( { query: queryState } );
```
Comment (JeanPaiva) [inline src/blocks/query-loop/components/InspectorControls.js:30]: This change was necessary to fix a bug that hold me for half day, lol.

Comment (tomusborne) [reply src/blocks/query-loop/components/InspectorControls.js:30]: This makes it so WP displays the confirm message when trying to leave the page if you've added the Query Loop even if nothing has changed. Anything we can do there?

File: src/components/post-type-records-select/index.js [hunk 2]
```diff
@@ -26,7 +36,7 @@ export default function PostTypeRecordsSelect( { postType, label, value, ...prop
 			isLoading={ isResolving }
 			isSearchable
 			isMulti
-			{ ...props }
+			{ ...otherProps }
```
Comment (tomusborne) [inline src/components/post-type-records-select/index.js:39]: How does this `otherProps` work?

File: src/blocks/query-loop/components/InspectorControls.js (hunk 1 above)
Comment (JeanPaiva) [reply src/blocks/query-loop/components/InspectorControls.js:30]: Working on a different solution

File: src/blocks/query-loop/components/utils.js [hunk 3]
```diff
@@ -23,8 +25,8 @@ function normalizeTaxQuery( taxQueryValue, isExclude = false ) {
 export function normalizeRepeatableArgs( query ) {
 	let normalizedQuery = normalizeArgs( query );
 
-	if ( Array.isArray( query.tax_query ) ) {
-		const normalizedTaxQuery = normalizeTaxQuery( query.tax_query );
+	if ( Array.isArray( normalizedQuery.tax_query ) ) {
```
[the remark this answers is not in the document]
Comment (JeanPaiva) [reply src/blocks/query-loop/components/utils.js:28]: Yes, the rest of the function was using `query` which was not normalized yet.

File: src/blocks/query-loop/components/InspectorControls.js (hunk 1 above)
Comment (tomusborne) [reply src/blocks/query-loop/components/InspectorControls.js:30]: What was the bug?

**Review by tomusborne — APPROVED** [review]

**Pull request merged**